### PR TITLE
Logging tolerations

### PR DESCRIPTION
**Pull request by mburke5678**
See comments in https://github.com/openshift/openshift-docs/pull/16422

### CODE REVIEW

File: logging/config/cluster-logging-tolerations.adoc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+include::modules/cluster-logging-kibana-tolerations.adoc[leveloffset=+1]
+
+include::modules/cluster-logging-curator-tolerations.adoc[leveloffset=+1]
+
+include::modules/cluster-logging-fluentd-tolerations.adoc[leveloffset=+1]
+
+[id="cluster-logging-tolerations-addtl-resources"]
+== Additional resources
+
+For more information on taints and tolerations, see
```
Comment (kalexand-rh) [inline logging/config/cluster-logging-tolerations.adoc]: s/on/about

File: modules/cluster-logging-curator-tolerations.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-curator.adoc
+
+[id="cluster-logging-curator-tolerations_{context}"]
+= Using tolerations to control the Curator pod placement
+
+You can control which node the Curator pod runs on
+using tolerations on the pod.
```
Comment (kalexand-rh) [inline modules/cluster-logging-curator-tolerations.adoc]: s/using/by using

File: modules/cluster-logging-curator-tolerations.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-curator.adoc
+
+[id="cluster-logging-curator-tolerations_{context}"]
+= Using tolerations to control the Curator pod placement
+
+You can control which node the Curator pod runs on
```
Comment (kalexand-rh) [inline modules/cluster-logging-curator-tolerations.adoc]: s/pod/Pod all places

File: modules/cluster-logging-curator-tolerations.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+instructs the node to repel all pods that do not tolerate the taint.
+
+
+.Prerequisites
+
+* Cluster logging and Elasticsearch must be installed.
+
+.Procedure
+
+. Use the following command to add a taint to a node where you want the Curator pod scheduled:
```
Comment (kalexand-rh) [inline modules/cluster-logging-curator-tolerations.adoc]: Should we add a step to view the available nodes and identify one to taint?

 s/the Curator pod scheduled/to schedule the Curator Pod

File: modules/cluster-logging-curator-tolerations.adoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+$ oc adm taint nodes <node-name> <key>=<value>:<effect>
+----
++
+For example:
++
+----
+$ oc adm taint nodes node1 key1=value1:NoExecute
+----
++
+This example places a taint on `node1` that has key `key1`, value `value1`, and taint effect `NoExecute`.
```
Comment (kalexand-rh) [inline modules/cluster-logging-curator-tolerations.adoc]: Is there more information about available taints and acceptable key=value pairs? I see what the taint's doing, but IDK the significance of the key=value pair.

I might emphasize that you must use the NoExecute taint to make this work right.

File: modules/cluster-logging-curator-tolerations.adoc [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        operator: "Exists"  <2>
+        effect: "NoExecute"  <3>
+        tolerationSeconds: 6000  <4>
+----
+<1> Specify the key you added to the node.
+<2> Specify the `Exists` operator to require the `key`/`value`/`effect` parameters to match. 
+<3> Specify the `NoExecute` effect.
+<4> Optionally, specify the `tolerationSeconds` parameter to set how long a pod can remain bound to a node before being evicted.
+
+This toleration matches the taint created by the `oc adm taint` command above. A pod this toleration would be able to schedule onto `node1`.
```
Comment (kalexand-rh) [inline modules/cluster-logging-curator-tolerations.adoc]: s/created/that is created

Not "above"
> ! above adj
Do not use to indicate a relative location in a document, as in “the above restrictions.” Use “previous” or
“preceding.”

File: modules/cluster-logging-elasticsearch-tolerations.adoc [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+. Use the following command to add taints and tolerations for cluster logging:
++
+----
+$ oc adm taint nodes <node-name> <key>=<value>:<effect>
+----
++
+For example:
++
+----
+$ oc adm taint nodes node1 key1=logging:NoExecute
```
Comment (kalexand-rh) [inline modules/cluster-logging-elasticsearch-tolerations.adoc]: Didn't we just do this? Do you put the pods on different nodes? Is there more guidance that we can share about how to direct them?

File: modules/cluster-logging-curator-tolerations.adoc [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  curation:
+    type: "curator"
+    curator:
+      tolerations: 
+      - key: "logging"  <1>
+        operator: "Exists"  <2>
+        effect: "NoExecute"  <3>
+        tolerationSeconds: 6000  <4>
+----
+<1> Specify the key you added to the node.
```
Comment (kalexand-rh) [inline modules/cluster-logging-curator-tolerations.adoc]: s/you added/that you added

File: modules/cluster-logging-fluentd-tolerations.adoc [hunk 9]
```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-fluentd.adoc
+
+[id="cluster-logging-fluentd-tolerations_{context}"]
+= Using tolerations to control the log collector pod placement
+
+You can ensure which nodes the logging collector pods run on
+using tolerations on the pods.
```
Comment (kalexand-rh) [inline modules/cluster-logging-fluentd-tolerations.adoc]: s/using/by using

File: modules/cluster-logging-fluentd-tolerations.adoc [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  operator: "Exists"
+----
+
+.Prerequisites
+
+* Cluster logging and Elasticsearch must be installed.
+
+.Procedure
+
+. Use the following command to add a taint to a node where you want logging collector pods scheduled:
```
Comment (kalexand-rh) [inline modules/cluster-logging-fluentd-tolerations.adoc]: s/logging collector pods scheduled/to schedule logging collector pods

File: modules/cluster-logging-kibana-tolerations.adoc [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+instructs the node to repel all pods that do not tolerate the taint.
+
+
+.Prerequisites
+
+* Cluster logging and Elasticsearch must be installed.
+
+.Procedure
+
+. Use the following command to add a taint to a node where you want Kibana pod scheduled:
```
Comment (kalexand-rh) [inline modules/cluster-logging-kibana-tolerations.adoc]: s/Kibana pod scheduled/to schedule the Kibana Pod

File: modules/cluster-logging-curator-tolerations.adoc [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+<2> Specify the `Exists` operator to require the `key`/`value`/`effect` parameters to match. 
+<3> Specify the `NoExecute` effect.
+<4> Optionally, specify the `tolerationSeconds` parameter to set how long a pod can remain bound to a node before being evicted.
+
+This toleration matches the taint created by the `oc adm taint` command above. A pod this toleration would be able to schedule onto `node1`.
+
```
Comment (kalexand-rh) [inline modules/cluster-logging-curator-tolerations.adoc]: I'd make the same suggestions for all of the following modules.

**Review by kalexand-rh — COMMENTED** [review]
Couple of nits and a request about planning the Pod placement.

File: modules/cluster-logging-elasticsearch-tolerations.adoc (hunk 7 above)
Comment (mburke5678) [reply modules/cluster-logging-elasticsearch-tolerations.adoc]: The RFE addresses a request to include the cluster logging components in the customer's broader toleration policies. The dev user story seems to emphasize preventing other workloads from running on the cluster logging nodes. 
I added wording to this effect.

**Pull request merged**